### ISSUE TITLE
[Hold #41442] [Guided Setup Stage 3][$500] Don't create default #announce rooms until a workspace has 3 or more members

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#34929]
## Problem
While we originally envisioned default rooms to streamline the a whole host of team collaboration use cases – including announcements to your employees, or chatting with other admins - as we've gained experience onboarding users to NewDot we've realized that this has had the effect of making onboarding harder. More specifically, your LHN starts out with a host of rooms and messages, though it's not clear which you should start with first:

 - Expensify (planned)
 - Concierge
 - Workspace chat
 - #admins
 - #announce 
 
 Even further, plenty of sign-ups are first and foremost looking for an expense and financial management app, and they're just one person, so this ends up confusing them that they've downloaded the wrong app, costing us valuable bottom-up chances that we'd otherwise keep around as long as sign-ups can effectively find what they're looking for. 
 
 ## Solution
Part of this solution is more targeted onboarding, and that's already being worked on in #wave9-collect-signups. Though equally, let's not assume group collaboration is what sign-ups are after – especially if they're the only member of the workspace. A good example is the #announce room, which is only relevant when a workspace has several members. 

Thus, in an effort to simplify what users are exposed to post-sign-up, and better target our functionality to their intended use case, let's:

 - Not create a default #announce room when the workspace is created
 - Instead, let's only create #announce rooms when the third member is added to the workspace (after all, otherwise two members could use a DM without issue). 
 
 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d20cdc896fd34ad2</li>
        <li>Upwork Job ID: 1755964549843316736</li>
        <li>Last Price Increase: 2024-02-16</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't create default #announce rooms until a workspace has 3 or more members
### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?
1. Remove [this](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1198C1-L1201C39) constants and everything depends on these  form [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192) and `buildOptimisticWorkspaceChats` utils
2. Create new `buildOptimisticAnounceChat` function in `ReportUtils` and transfer  [this](https://github.com/Expensify/App/blob/85c990eb3ddba4a8709ce51f69d5c1393896f1ea/src/libs/ReportUtils.ts#L3450C1-L3468C7) part of code to there
3. Create new `createAnnounceChat` function similar to [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192C10-L1192C25). E.g Below
```
function createAnnounceChat( policyName, policyID ) {
    const {
        announceChatReportID,
        announceChatData,
        announceReportActionData,
        announceCreatedReportActionID,

    } = ReportUtils.buildOptimisticAnounceChat(policyID, workspaceName);

    const optimisticData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`,
            value: {
                [sessionAccountID]: {
                    role: CONST.POLICY.ROLE.ADMIN,
                    errors: {},
                },
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
                },
                ...announceChatData,
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: announceReportActionData,
        },
    ];

    const successData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: null,
                },
                pendingAction: null,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: {
                [announceCreatedReportActionID]: {
                    pendingAction: null,
                },
            },
        },
    ];

    const failureData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: null,
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: null,
        },
    ];

    const params = {
        policyID,
        announceChatReportID,
        announceCreatedReportActionID,
    };

    API.write(WRITE_COMMANDS.CREATE_WORKSPACE, params, {optimisticData, successData, failureData});
}
```
4. Add below condition in [addMembersToWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L610C10-L610C31) function
```
    if(allPolicyMembers?.[membersListKey]){
        createAnounceChat(policiName, policyID)
    }
``` 

### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't create default #announce rooms until a workspace has 3 or more members
### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?
1. Remove [this](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1198C1-L1201C39) constants and everything depends on these  form [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192) and [createWorkspaceFromIOUPayment](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1555C10-L1555C39)
2. Create new `buildOptimisticAnounceChat` function in `ReportUtils` and move [this](https://github.com/Expensify/App/blob/85c990eb3ddba4a8709ce51f69d5c1393896f1ea/src/libs/ReportUtils.ts#L3450C1-L3468C7) part of the code there.
3. Create new `createAnnounceChat` function similar to [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192C10-L1192C25). E.g Below
4. Create new `createAnnounceChat` endpoint in  backend side 
```
function createAnnounceChat( policyName, policyID ) {
    const {
        announceChatReportID,
        announceChatData,
        announceReportActionData,
        announceCreatedReportActionID,

    } = ReportUtils.buildOptimisticAnounceChat(policyID, policyName);

    const optimisticData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`,
            value: {
                [sessionAccountID]: {
                    role: CONST.POLICY.ROLE.ADMIN,
                    errors: {},
                },
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
                },
                ...announceChatData,
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: announceReportActionData,
        },
    ];

    const successData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: null,
                },
                pendingAction: null,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: {
                [announceCreatedReportActionID]: {
                    pendingAction: null,
                },
            },
        },
    ];

    const failureData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: null,
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: null,
        },
    ];

    const params = {
        policyID,
        announceChatReportID,
        announceCreatedReportActionID,
    };

    API.write("createAnnounceChat", params, {optimisticData, successData, failureData});
}
```
5. Add below condition in [addMembersToWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L610C10-L610C31) function
```
    if(allPolicyMembers?.[membersListKey]?.length > 2){
        createAnounceChat(policiName, policyID)
    }
``` 

### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't create default #announce rooms until a workspace has 3 or more members
### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?
1. Remove [this](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1198C1-L1201C39) constants and everything depends on these  form [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192) and [createWorkspaceFromIOUPayment](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1555C10-L1555C39)
2. Create new `buildOptimisticAnounceChat` function in `ReportUtils` and move [this](https://github.com/Expensify/App/blob/85c990eb3ddba4a8709ce51f69d5c1393896f1ea/src/libs/ReportUtils.ts#L3450C1-L3468C7) part of the code there.
3. Create new `createAnnounceChat` endpoint in  backend side 
4. Create new `createAnnounceChat` function similar to [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192C10-L1192C25). E.g Below
```
function createAnnounceChat( policyName, policyID ) {
    const {
        announceChatReportID,
        announceChatData,
        announceReportActionData,
        announceCreatedReportActionID,

    } = ReportUtils.buildOptimisticAnounceChat(policyID, policyName);

    const optimisticData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`,
            value: {
                [sessionAccountID]: {
                    role: CONST.POLICY.ROLE.ADMIN,
                    errors: {},
                },
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
                },
                ...announceChatData,
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: announceReportActionData,
        },
    ];

    const successData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: null,
                },
                pendingAction: null,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: {
                [announceCreatedReportActionID]: {
                    pendingAction: null,
                },
            },
        },
    ];

    const failureData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: null,
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: null,
        },
    ];

    const params = {
        policyID,
        announceChatReportID,
        announceCreatedReportActionID,
    };

    API.write("createAnnounceChat", params, {optimisticData, successData, failureData});
}
```
5. Add below condition in [addMembersToWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L610C10-L610C31) function
```
    if(allPolicyMembers?.[membersListKey]?.length > 2){
        createAnounceChat(policiName, policyID)
    }
``` 

### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't create default #announce rooms until a workspace has 3 or more members
### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?
1. Remove [this](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1198C1-L1201C39) constants and everything depends on these  form [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192) and [createWorkspaceFromIOUPayment](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1555C10-L1555C39)
2. Create new `buildOptimisticAnounceChat` function in `ReportUtils` and move [this](https://github.com/Expensify/App/blob/85c990eb3ddba4a8709ce51f69d5c1393896f1ea/src/libs/ReportUtils.ts#L3450C1-L3468C7) part of the code there.
3. Create new `CreateAnnounceChat` endpoint in  backend side 
4. Create new `createAnnounceChat` function similar to [createWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L1192C10-L1192C25). E.g Below
```
function createAnnounceChat( policyName, policyID ) {
    const {
        announceChatReportID,
        announceChatData,
        announceReportActionData,
        announceCreatedReportActionID,

    } = ReportUtils.buildOptimisticAnounceChat(policyID, policyName);

    const optimisticData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`,
            value: {
                [sessionAccountID]: {
                    role: CONST.POLICY.ROLE.ADMIN,
                    errors: {},
                },
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
                },
                ...announceChatData,
            },
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: announceReportActionData,
        },
    ];

    const successData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: {
                pendingFields: {
                    addWorkspaceRoom: null,
                },
                pendingAction: null,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: {
                [announceCreatedReportActionID]: {
                    pendingAction: null,
                },
            },
        },
    ];

    const failureData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
            value: null,
        },
        {
            onyxMethod: Onyx.METHOD.SET,
            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${announceChatReportID}`,
            value: null,
        },
    ];

    const params = {
        policyID,
        announceChatReportID,
        announceCreatedReportActionID,
    };

    API.write("CreateAnnounceChat", params, {optimisticData, successData, failureData});
}
```
5. Add below condition in [addMembersToWorkspace](https://github.com/Expensify/App/blob/6d302160b998fdd10f440998c3c9e19a4c1d7ca3/src/libs/actions/Policy.ts#L610C10-L610C31) function
```
    if(allPolicyMembers?.[membersListKey]?.length === 2){
        createAnounceChat(policiName, policyID)
    }
``` 

### What alternative solutions did you explore? (Optional)

**Comment by quinthar** [comment]
I think this would be done on the back end, wouldn't it be?  We would create the announce room when the 3rd person is added to the workspace, which is something we should probably do inside auth given that there are so many ways to invite someone.

**Comment by allroundexperts** [comment]
Yes, I think we need to handle this on the frontend as well when creating a workspace optimistically. Although from the proposals above, I don't understand why do we need to create a new `CreateAnnounceChat` endpoint on the backend. The `WRITE_COMMANDS.CREATE_WORKSPACE` endpoint should just not send the announce chat data if workspace has less than 3 members from the backend.

**Comment by allroundexperts** [comment]
We should just update the optimistic data. Rest can be handled by the backend.

**Comment by anmurali** [comment]
From a quick discussion in Slack, I think we want to make this change backend. Members can be added from several different front end flows and so it would be cleaner to do it once, backend rather than scout down every front end flow to address

**Comment by allroundexperts** [comment]
@anmurali We need to do the frontend part as well in order to make the optimistic data work.

**Comment by allroundexperts** [comment]
@rayane-djouah Your proposal is still not making a lot of sense to me. I think its too complex and has un-needed changes. All we need to do on the frontend is to make sure that the announce room is not created optimistically if members are less than 3.

**Comment by allroundexperts** [comment]
> @allroundexperts could you please elaborate more on what changes do you think that are not needed?

I think you're creating multiple workspace creation requests.

**Comment by allroundexperts** [comment]
> @allroundexperts can we create the announce room optimistically by using `Onyx.set()` and without calling any API endpoint ?
We are already calling the endpoint in the `createWorkspace` action. It's not needed to call it twice.

**Comment by allroundexperts** [comment]
@rojiphil's [proposal](https://github.com/Expensify/App/issues/34929#issuecomment-1945650801) looks good to me. It handles the case where a user is added later on after a workspace gets created.

🎀 👀 🎀 C+ reviewed

**Comment by shahinyan11** [comment]
@allroundexperts Don't you have any feedback on my proposal?

**Comment by allroundexperts** [comment]
@shahinyan11 It looks pretty much the same as what @rayane-djouah first proposed. As such, It has the same issues which I explained above.

**Comment by shahinyan11** [comment]
> @shahinyan11 It looks pretty much the same as what @rayane-djouah first proposed. 
@allroundexperts He didn't propose first, he just updated his proposal more then 10 times  after mine

> As such, It has the same issues which I explained above.
And this is understandable

**Comment by NikkiWines** [comment]
Nice, agreed @rojiphil's proposal looks solid 👍

**Comment by rojiphil** [comment]
> We will need BE support for creation of `#announce` room through `AddMembersToWorkspace` request. Further, BE should not create `#announce` room chat report on workspace creation.

Thanks @NikkiWines for the assignment here. 
This issue also needs BE support as mentioned above. 
Please let me know when BE is ready and I will raise the PR in FE.

**Comment by NikkiWines** [comment]
Started on the backend changes today! I'll post here when the changes are deployed!

**Comment by quinthar** [comment]
So is this entirely back end -- is there any front end component?

**Comment by rojiphil** [comment]
> So is this entirely back end -- is there any front end component?

Not entirely BE. There are FE changes too.

**Comment by johncschuster** [comment]
Thanks for the clarification, @rojiphil! @NikkiWines, can you link your PR so we can follow along!

**Comment by NikkiWines** [comment]
Sure thing! This change ends up impacting more logic than I anticipated and we have a bunch of tests to adjust - sorry for the delay 🙇 

The auth PR is [here](https://github.com/Expensify/Auth/pull/9990). We'll also need a corresponding web PR which I'll link shortly.

**Comment by johncschuster** [comment]
Awesome. Thanks, @NikkiWines!

**Comment by NikkiWines** [comment]
Been focused on a wave6 issue so haven't had time to update the backend PR but I'll try to prioritize this during the week (or will hand it off otherwise)

**Comment by johncschuster** [comment]
Thanks for the update, @NikkiWines!

**Comment by johncschuster** [comment]
@NikkiWines how's this one coming along? Were you able to make space for this one last week?

**Comment by NikkiWines** [comment]
I did! Still working through some of the web tests but made progress and should hopefully have both PRs in review by tomorrow

**Comment by johncschuster** [comment]
Bumping to keep melvin happy while we wait on the PRs above

**Comment by NikkiWines** [comment]
Sorry for the delay - there were a lot (32) of different tests that involved the existence of the #announce room  in some way and we had to adjust some of the onyx update logic as well. All the associated PRs are in review and should hopefully be merged/deployed soon.

**Comment by johncschuster** [comment]
Thanks for the update, @NikkiWines!

**Comment by johncschuster** [comment]
Bumping to `Weekly` while we wait on the PRs to be merged.

**Comment by NikkiWines** [comment]
Per [this discussion](https://expensify.slack.com/archives/C066HJM2CAZ/p1710355809129009) I think we should put this on hold until https://github.com/Expensify/Web-Expensify/pull/40764 is merged

**Comment by johncschuster** [comment]
Looks like https://github.com/Expensify/Web-Expensify/pull/40764 was recently merged. Taking this off hold!

**Comment by NikkiWines** [comment]
Sorry, this is actually held on https://github.com/Expensify/Web-Expensify/pull/41442 - which is still on hold (waiting for https://github.com/Expensify/Auth/pull/10021) - so this isn't ready to come off hold yet.

**Comment by johncschuster** [comment]
No worries! I'll keep an eye on that one and take us off hold when it's merged!

**Comment by johncschuster** [comment]
Nothing new to report here. It looks like we're still waiting on [these](https://github.com/Expensify/App/issues/34929#issuecomment-2052595831)

**Comment by johncschuster** [comment]
@NikkiWines, it looks like https://github.com/Expensify/Auth/pull/10021 was merged and https://github.com/Expensify/Web-Expensify/pull/41442 was closed. Where does that put this issue?

**Comment by NikkiWines** [comment]
Oh yes, thanks for the bump. I had something written out but hadn't sent it 😅

Since #vip-vsb is currently paused, I've closed those PRs (but kept the branches so the work is retained) so they're not unnecessarily clogging up the reviewers' K2s. For this issue, let's unassign anyone non-internal and remove the `Monthly` label until we-re reprioritizing #vip issues tasks?

**Comment by shawnborton** [comment]
Should we consider moving this one over to #wave-collect then? cc @JmillsExpensify @trjExpensify @danielrvidal for thoughts. I only ask because I think this one will still impact a lot of signups as we start to push new sign ups to be on NewDot.

**Comment by trjExpensify** [comment]
Yeah, maybe @danielrvidal wants to put it under the `Guided Setup: Stage 3` project as part of readiness for viral actors invited to NewDot. It has an impact on what they see in the LHN when invited.

**Comment by trjExpensify** [comment]
Cool, so as per here, we will put this under the Guided Setup: Stage 3 project. I can put this in Hot Picks, but @NikkiWines it would be useful for a rundown of what's left to do or pick up from where you left off with the work from [here](https://github.com/Expensify/Web-Expensify/pull/41442).

**Comment by NikkiWines** [comment]
Nice, I'll be finishing up with #wave-collect Sage Intacct stuff shortly so happy to stay on this if it needs a volunteer. 

As for a summary - there are changes needed in Web and Auth to move this forward. We were initially held up on https://github.com/Expensify/Auth/pull/10021 because that moved the onyxUpdates for `SharePolicy`  from Web to Auth. When we share a policy, we can bump the number of members to over 3, impacting whether or not we create an #announce room - hence the hold-up. 

The bulk of the changes needed for this are in the [auth PR](https://github.com/Expensify/Auth/pull/9990/files#diff-72abd4ce180c1fbdf93258363f1da3647a418fabc8240ef6d6d8389f17d998e7). The core logic is already implemented from earlier, but we'll need to adjust the onyx update logic (that we were originally doing in the web PR) accordingly, 

The [Web PR](https://github.com/Expensify/Web-Expensify/pull/41442/files) now will likely be much smaller but it'll still be necessary as we have some integrations tests that check for the #announce room creation immediately after a policy is created for the first time. In fact, it might be easier to just start fresh on the web side, since the bulk of the changes that were being made there are now being made in Auth

**Comment by johncschuster** [comment]
This is currently on hold. Not overdue.

**Comment by johncschuster** [comment]
@NikkiWines bump on the above question!

**Comment by NikkiWines** [comment]
Sorry, lost track of this one as I was still working on Intacct release issues. I'll update the auth and web PRs this week.

**Comment by NikkiWines** [comment]
PR is being updated - looks good overall just adjusting the tests for the onyx updates

**Comment by NikkiWines** [comment]
PR is in review - auth tests are fixed, waiting on one possible fix in a different PR for web-e tests

**Comment by anmurali** [comment]
@NikkiWines is there an update on this?

**Comment by NikkiWines** [comment]
Tests are all passing now, waiting on reviews before we merge this. Note for myself to notify Applause about this change since it'll impact a bunch of our regression tests around checking for the existence of an #announce room

**Comment by trjExpensify** [comment]
Executing the CVP, and checking in here on the next steps. The Auth PR hit prod yesterday, are we on to a final App PR now?

**Comment by NikkiWines** [comment]
Since the room won't exist until the 3rd user is added, I'm not sure we need a corresponding app PR unless we want to do something particular when the #announce room is eventually created

**Comment by trjExpensify** [comment]
Oh cool, so we're good to go here then!

**Comment by NikkiWines** [comment]
Yep, I believe so! Closing this out

**Comment by NikkiWines** [comment]
Ooh, although we may be optimistically trying to create the announce room so actually App changes might be required - let me do some testing .

**Comment by rojiphil** [comment]
> Ooh, although we may be optimistically trying to create the announce room so actually App changes might be required - let me do some testing .

That's correct @NikkiWines. Here is a test video that demonstrates the need for FE changes. 

https://github.com/user-attachments/assets/4b31c1ad-698c-4c78-af80-7fc11f4f48be

**Comment by rojiphil** [comment]
Working on the App PR now

**Comment by rojiphil** [comment]
@NikkiWines I have created a draft PR that addresses most of the issues, but we need to agree on how FE can optimistically generate the `announce` room. And, at the same time, the BE also seamlessly considers that without resulting in two #announce rooms in FE. The following test video demonstrates the problem for the draft PR. 

Can BE handle this if we pass announce room chat data along with `AddMembersToWorkspace` request and also additionally pass the params `announceChatReportID` and `announceCreatedReportActionID` [here](https://github.com/Expensify/App/blob/e7bd29ad4a3f6cb4429fb383404f266791d42fa9/src/libs/actions/Policy/Member.ts#L632C1-L634C18) as we do for `createWorkspaceChat` [here](https://github.com/Expensify/App/blob/e7bd29ad4a3f6cb4429fb383404f266791d42fa9/src/libs/actions/Policy/Policy.ts#L1854C1-L1861C39) or do you see any other approach to resolve this?


https://github.com/user-attachments/assets/9d51cef7-0b1e-4350-9c38-3be5f65bd230

**Comment by anmurali** [comment]
@NikkiWines - can we take care of this issue? It would be really great to get this done as it is what stands between Stage 2 onboarding (within the scope of which we said we would clean this up) being actively worked on vs. done.

**Comment by NikkiWines** [comment]
> Can BE handle this if we pass announce room chat data along with AddMembersToWorkspace request and also additionally pass the params announceChatReportID and announceCreatedReportActionID [here](https://github.com/Expensify/App/blob/e7bd29ad4a3f6cb4429fb383404f266791d42fa9/src/libs/actions/Policy/Member.ts#L632C1-L634C18) as we do for createWorkspaceChat [here](https://github.com/Expensify/App/blob/e7bd29ad4a3f6cb4429fb383404f266791d42fa9/src/libs/actions/Policy/Policy.ts#L1854C1-L1861C39) or do you see any other approach to resolve this?

Yes, I think it makes sense to follow what we did for `createWorkspaceChat`, and I can update the backend for  `AddMembersToWorkspace` to take in the `announceChatReportID` and `announceCreatedReportActionID`.

**Comment by rojiphil** [comment]
> Yes, I think it makes sense to follow what we did for `createWorkspaceChat`, and I can update the backend for `AddMembersToWorkspace` to take in the `announceChatReportID` and `announceCreatedReportActionID`.

Thanks and I have updated the draft PR to send across `announceChatReportID` and `announceCreatedReportActionID` as params in `AddMembersToWorkspace` request. I look forward to testing this out once BE is ready.

**Comment by NikkiWines** [comment]
The backend PRs ([one](https://github.com/Expensify/Auth/pull/12444), and [two](https://github.com/Expensify/Web-Expensify/pull/43455)) are up and in review

**Comment by NikkiWines** [comment]
The [Auth PR](https://github.com/Expensify/Auth/pull/12444) has been merged, once that's deployed we can merge the [Web PR](https://github.com/Expensify/Web-Expensify/pull/43455). And when the Web PR is live then we can get https://github.com/Expensify/App/pull/48660 over the line

**Comment by NikkiWines** [comment]
[Web PR](https://github.com/Expensify/App/issues/34929) has been merged - should go to prod by tuesday.

**Comment by NikkiWines** [comment]
Web-E PR is on staging now!

**Comment by NikkiWines** [comment]
Backend changes are live, we should be able to move forward with the [App PR](https://github.com/Expensify/App/pull/48660) now

**Comment by rojiphil** [comment]
> Backend changes are live, we should be able to move forward with the [App PR](https://github.com/Expensify/App/pull/48660) now

Thanks @NikkiWines  and yeah, BE is full of life and works like a charm too.
PR is ready for review @allroundexperts

**Comment by madmax330** [comment]
@JmillsExpensify there's an issue here: https://github.com/Expensify/Expensify/issues/427525

For invoicing we expect the announce room to exist, but when you send an invoice to someone and they chose to pay as a business, we create a workspace for them where they are the only member, so the annouce room doesn't get created (since they're the only memeber)

I'm not sure what we should do in this case. Should we:
1. create the announce room with < 3 members for this case
2. use another room for invoicing
3. something else?

**Comment by trjExpensify** [comment]
What's the announce room used for in that invoice case, @madmax330?

**Comment by madmax330** [comment]
Apparently nothing 😄 
Or if something I can't find it. 
But I think it's mainly because we create the room optimistically, so when the API doesn't create it it causes a bug on the front-end. So maybe we should just stop creating it altogether?

**Comment by anmurali** [comment]
https://fsty.io/v/6X2rBN3K

I have seen this in about a dozen sessions. The #announce room shows up but when clicked shows a not here error. Is this because the room is created optimistically? I think we should just stop creating it. Create it when there are 3+ members in the workspace.

**Comment by trjExpensify** [comment]
That's already the plan, isn't it? The App PR is in review: https://github.com/Expensify/App/pull/48660

**Comment by danielrvidal** [comment]
I'd vote we stop creating it altogether. I saw this in 3 full stories impacting customers trying to adopt. Can we prioritize as CRITICAL?

**Comment by trjExpensify** [comment]
Yeah, I think we still need this App PR to be merged: https://github.com/Expensify/App/pull/48660

**Comment by JmillsExpensify** [comment]
Agreed. Can we update that PR to CRITICAL, mostly because it's affecting production users. What do ya'll think @NikkiWines @allroundexperts?

**Comment by trjExpensify** [comment]
Ah yeah, I commented in the PR already. I've escalated it to slack here: https://expensify.slack.com/archives/C02NK2DQWUX/p1727347525853669

**Comment by NikkiWines** [comment]
This can be closed I think - https://github.com/Expensify/App/pull/48660 has been on prod for a few weeks now - not sure why the issue wasn't updated with the payment details. 

@johncschuster can we issue payment to @rojiphil and @allroundexperts for the PR and C+ review - thank you 🙇

**Comment by johncschuster** [comment]
# Payment Summary:

Contributor: @rojiphil paid $500 via [Upwork](https://www.upwork.com/nx/wm/offer/101005550) - PAID! 🎉 

Contributor+: @allroundexperts due $500 via NewDot

Upwork job [here](https://www.upwork.com/nx/wm/offer/101005550)! Please apply

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
